### PR TITLE
[TG-3495] Feature/java/support java6

### PR DESCRIPTION
**Pull request by mgudemann**

### CODE REVIEW

**Comment by mgudemann** [comment]
there is diffblue/test-gen#1799 that bumps the deeptest-utils pointer to include this Java6 build and has a modified `pom.xml` generator to check Java6 source and target.

File: pom.xml [hunk 1]
```diff
@@ -171,8 +171,8 @@
             <artifactId>maven-compiler-plugin</artifactId>
             <version>3.1</version>
             <configuration>
-              <source>1.8</source>
-              <target>1.8</target>
+              <source>1.6</source>
```
Comment (thk123) [inline pom.xml]: I guess it is forward compatible, but I think it would be best if we can compile against all 3 supported Java versions on travis

**Comment by thk123** [comment]
In fact specifically here some cases are documented where valid java 7 code doesn't compile with Java 8: http://www.oracle.com/technetwork/java/javase/8-compatibility-guide-2156366.html#A999198

**Comment by mgudemann** [comment]
@thk123 compiling with openJDK 7/8 and oracle JDK 8 now. oracle 7 isn't available anymore it seems and openjdk6 doesn't work due to dependencies having class file version 51.0

File: pom.xml [hunk 2]
```diff
@@ -138,11 +138,57 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>jdk6build</id>
```
[the remark this answers is not in the document]
Comment (mgudemann) [reply pom.xml]: added comment
```
        <!-- this selects any Java7 JDK or higher, split was needed as
             checkstyle is incompatible with JDK6 -->
``` and moved after default profile

File: pom.xml [hunk 3]
```diff
@@ -138,11 +138,57 @@
...
+        </plugins>
+      </build>
+    </profile>
     <profile>
       <id>stdbuild</id>
       <activation>
-        <activeByDefault>true</activeByDefault>
+        <jdk>[1.7,1.9)</jdk>
       </activation>
+      <!-- <activation> -->
```
[the remark this answers is not in the document]
Comment (mgudemann) [reply pom.xml]: isn't removed now

File: .travis.yml [hunk 4]
```diff
@@ -1 +1,29 @@
...
+matrix:
+  include:
+  - jdk: openjdk6
+    env: CUSTOM_MVN_VERION="3.2.5"
+  - jdk: openjdk7
+  - jdk: openjdk8
+  - jdk: oraclejdk8
+
+install:
+  - if [[ -n "${CUSTOM_MVN_VERION}" ]]; then
```
[the remark this answers is not in the document]
Comment (mgudemann) [reply .travis.yml]: added  a comment here

File: .travis.yml [hunk 5]
```diff
@@ -1 +1,29 @@
...
+      echo "Download Maven ${CUSTOM_MVN_VERION}....";
+      if [[ "${CUSTOM_MVN_VERION}" == "3.0" ]]; then
+        wget https://archive.apache.org/dist/maven/binaries/apache-maven-3.0-bin.zip || travis_terminate 1;
```
[the remark this answers is not in the document]
Comment (mgudemann) [reply .travis.yml]: came from the original implementation I found, is removed, the URL it came from is linked, too

**Pull request merged**